src/julia/Integer.jl

```diff
@@ -28,6 +28,8 @@ is_exact_type(::Type{T}) where T <: Integer = true
 
 is_domain_type(::Type{T}) where T <: Integer = true
 
+base_ring(::Vector{T}) where T <: Integer = T
```

Comment:
This seems not what `base_ring` is supposed to do. I would be in favor of removing this if it does not break downstsream

Reply:
It is used in Hecke. There is some code that should work for vectors of integers both as `Vector{Int}` and `ZZMatrix`. Maybe we should introduce `_base_ring` for this, I don't know.

Reply:
What is it used for? And why is it just for integer vectors? E.g. this gives an error:
```
julia> R,t = QQ[:t] ; base_ring([t])
ERROR: MethodError: no method matching base_ring(::Vector{QQPolyRingElem})
The function `base_ring` exists, but no method is defined for this combination of argument types.
```

This makes it even stranger that `base_ring` exists for e.g. `Vector{Int}`. Really seems to me that Hecke should use a differently named helper here?

Reply:
If at all I'd expect `base_ring` on a vector to return the parent of the elements of the vector, which would suggest
```suggestion
base_ring(v::Vector{T}) where T = parent(v[1])
```
or if one just wants the type, then perhaps
```suggestion
base_ring_type(v::Vector{T}) where T = parent_type(T)
```

Reply:
@thofma This is used in the Plesken--Souvignier lattice stuff. What do you think?